docs/okeps/okep-5296-bgp.md

```diff
@@ -0,0 +1,663 @@
+* As a user where maximum throughput is a priority, I want to reduce packet overhead by not having to encapsulate
+  traffic with Geneve.
+* As an egress IP user, I do not want to have to restrict my nodes to the same layer 2 segment and prefer
+  to use a pure routing implementation to handle advertising egress IP movement across nodes.
+
+## Proposed Solution
+
+OVN-Kubernetes will leverage other projects that already exist to enable BGP in Linux. FRR will be used as the BGP
+speaker, as well as provide support for other protocols like BFD. For FRR configuration, the
+MetalLB project has already started an API to be able to configure FRR:
+[https://github.com/metallb/frr-k8s](https://github.com/metallb/frr-k8s). While some of the configuration support for
+FRR may be directly exposed by FRR-K8S API, it may also be the case that some intermediary CRD provided by
+OVN-Kubernetes is required to integrate OVN-Kubernetes networking concepts into FRR.
+
+Functionally, FRR will handle advertising and importing routes and configuring those inside a Linux VRF. OVN-Kubernetes
+will be responsible for listening on netlink and configuring logical routers in OVN with routes learned
+by FRR. OVN-Kubernetes will manage FRR configuration through the FRR-K8S API in order to advertise routes outside of the
+node onto the provider network.
+
+There should be no changes required in FRR. FRR-K8S may need extended APIs to cover the OVN-Kubernetes use cases
+proposed in this enhancement. OVN will require no changes.
+
+### Workflow Description
+
+An admin has the ability to configure BGP peering and choose what networks to advertise. A tenant is able to define
+networks for their namespace, but requires admin permission in order to expose those networks onto the provider's BGP
+fabric. A typical workflow will be for a user or admin to create a user-defined network, and then the admin will be
+responsible to:
+
+1. If setting up VRF-Lite, do any host modifications necessary via NMState to enslave an IP interface to the
+   matching network VRF.
+2. Configure BGP peering via interacting with the FRR-K8S API for a given set of worker nodes. Also define filters for
+   what routes should be received from the provider network.
+3. Create the OVN-Kubernetes RouteAdvertisements CR to configure what routes and where to advertise them.
+4. Verify peering and that routes have been propagated correctly.
+5. If desired, change the transport type from geneve on the network to none in order to no longer use an overlay.
+
+For detailed examples, see the [BGP Configuration](#bgp-configuration) section.
+
+### API Extensions
+FRR-K8S API will be used in order to create BGP Peering and configure other BGP related configuration. A
+RouteAdvertisements CRD will be introduced in order to determine which routes should be advertised for specific networks.
+Additionally, the network CRD will be modified in order to expose a new transport field to determine if encapsulation
```

Comment:
should we link to the no-overlay mode enhancement PR here? or remove it from this OKEP since we didn't implement it?